src/Campaigns/Migrations/MigrateFormsToCampaignForms.php

```diff
@@ -38,57 +38,99 @@ public static function timestamp(): int
      */
     public function run()
     {
-        foreach(DonationForm::query()->getAll() ?? [] as $form) {
-            $this->createParentCampaignForDonationForm($form);
-        }
+        DB::transaction(function() {
+            try {
+                foreach($this->getFormData() as $formData) {
+                    $this->createParentCampaignForDonationForm($formData);
+                }
+            } catch (DatabaseQueryException $exception) {
+                DB::rollback();
+                throw new DatabaseMigrationException('An error occurred while creating initial campaigns', 0, $exception);
+            }
+        });
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function getFormData(): array
+    {
+        return DB::table('posts', 'forms')
+            ->select(
+                ['ID', 'id'],
+                ['post_title', 'title'],
+                ['post_status', 'status'],
+                ['post_date', 'createdAt'],
+                ['meta_value', 'settings']
+            )
+            ->join(function (JoinQueryBuilder $builder) {
+                $builder
+                    ->leftJoin('give_formmeta', 'formmeta')
+                    ->on('formmeta.form_id', 'forms.ID');
+            })
+            ->where('forms.post_type', 'give_forms')
+            ->where('formmeta.meta_key', 'formBuilderSettings')
+            ->getAll();
     }
 
     /**
      * @unreleased
      */
-    public function createParentCampaignForDonationForm(DonationForm $form)
+    public function createParentCampaignForDonationForm($formData): void
     {
-        $campaign = Campaign::create([
-            'type' => CampaignType::CORE(),
-            'title' => $form->title,
-            'shortDescription' => $form->settings->formExcerpt,
-            'longDescription' => $form->settings->description,
-            'logo' => $form->settings->designSettingsLogoUrl,
-            'image' => $form->settings->designSettingsImageUrl,
-            'primaryColor' => $form->settings->primaryColor,
-            'secondaryColor' => $form->settings->secondaryColor,
-            'goal' => (int) $form->settings->goalAmount,
-            'status' => $this->mapFormStatusToCampaignStatus($form->status),
-            'startDate' => new DateTime($form->settings->goalStartDate),
-            'endDate' => new DateTime($form->settings->goalEndDate),
-        ]);
+        $formId = $formData->id;
+        $formTitle = $formData->title;
+        $formStatus = $formData->status;
+        $formCreatedAt = $formData->createdAt;
+        $formSettings = json_decode($formData->settings);
+
+        $campaignId = DB::table('give_campaigns')
+            ->insert([
+                'campaign_type' => CampaignType::CORE()->getValue(),
```

Comment:
@kjohnson Even this, I would put in the actual value and not reference the VO, as we'd potentially have a migration later that changes this value, moves the VO, etc.

Reply:
Resolved, see [bde30e0](https://github.com/impress-org/givewp/pull/7532/commits/bde30e049132c1d485ff58253ffd30d2b2044ff8).